add Annotation

huangdeyin opened the pull request:
My mysql database table name does not use plural form. Now the generated files always need me to manually add annotations, which seems too inconvenient. I have solved this problem, but I am not sure whether it is the most good implementation
like this
![image](https://user-images.githubusercontent.com/18435236/150676180-08a6475b-8279-48c4-b946-501326a0b831.png)

await...

zeevmoney commented:
Hi @huangdeyin,

Thank you for the contribution, while this is a needed feature, it can't be automatic.
What I had in mind  is some sort of flag, config file or interactive prompt.

I will keep this PR open for some time in order to discuss it, please let me know what do you think.

huangdeyin commented:
Thank you very much for your suggestion. I refer to your comments and change it to an optional configuration item through command line parameters. On my computer, it has worked, but since I don’t know the process of gotest, if I need to complete the test Use case, that may take more time, looking forward to your correction

usecase
![image](https://user-images.githubusercontent.com/18435236/150713676-c8b867a8-5683-4905-a988-e56f9c5b5864.png)

a8m reviewed (commented):
Hey @huangdeyin 👋 

In my opinion, the code that inspects the tables today should add the `entsql.Table` annotation if the table-name that was used in the inspection is different from the format ent uses.

That means, the CLI option is not needed, and you only need to add this if-check on `upsertNode`.

a8m commented:
Let me know if you need help in ent Discord channel and I can guide you on how to fix this.

huangdeyin commented:
> 嘿@huangdeyin 👋
> 
> `entsql.Table`在我看来，如果在检查中使用的表名与 ent 使用的格式不同，那么今天检查表的代码应该添加注释。
> 
> 这意味着，不需要 CLI 选项，您只需添加此 if-check on `upsertNode`。

ok

huangdeyin commented:
> Hey @huangdeyin 👋
> 
> In my opinion, the code that inspects the tables today should add the `entsql.Table` annotation if the table-name that was used in the inspection is different from the format ent uses.
> 
> That means, the CLI option is not needed, and you only need to add this if-check on `upsertNode`.

ok, i saw your comment, let me try and very sorry for replying so late because China is celebrating new year

huangdeyin commented:
@zeevmoney 
I modified the implementation code, removed the CLI option for annotations, and now it can automatically decide whether to add annotations based on the table name
I added new test cases and it's working fine on my computer now.
![image](https://user-images.githubusercontent.com/18435236/152931588-1e05d681-d250-41ee-90b8-e7b74c8a022a.png)

@a8m 
Your suggestion is valid, there may be more non-default items that need to be added in the future, but at present I don't have a real scene to reproduce them

If you find any problem, let me know in the comments and I will try to solve it in my spare time 
Thanks a lot for your suggestions

a8m commented on .gitignore:
```diff
@@ -0,0 +1 @@
+.idea
```
Can you please keep this in your global gitignore?

https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer

a8m commented on internal/entimport/import.go:
```diff
@@ -247,10 +249,17 @@ func upsertNode(field fieldFunc, table *schema.Table) (*schemast.UpsertSchema, e
 	for _, f := range upsert.Fields {
 		fields[f.Descriptor().Name] = f
 	}
+
```
rm empty line

a8m commented on internal/entimport/import.go:
```diff
@@ -259,6 +268,7 @@ func upsertNode(field fieldFunc, table *schema.Table) (*schemast.UpsertSchema, e
 		fields[pk.Descriptor().Name] = pk
 		upsert.Fields = append(upsert.Fields, pk)
 	}
+
```
rm empty line

a8m commented on internal/entimport/import.go:
```diff
@@ -247,10 +249,17 @@ func upsertNode(field fieldFunc, table *schema.Table) (*schemast.UpsertSchema, e
 		Name: typeName(table.Name),
 	}
+	if tableName(table.Name) != table.Name {
+		annotationsConfig := entsql.Annotation{}
+		annotationsConfig.Table = table.Name
+		upsert.Annotations = []entschema.Annotation{annotationsConfig}
```
```suggestion
		upsert.Annotations = []entschema.Annotation{
			entsql.Annotation{Name: table.Name},
		}
```

a8m reviewed (approved):
Amazing @huangdeyin! Thanks for the contribution. 

I'll let @zeevmoney also to give it a look, but please fix the minor style comments and I think we can merge this.

huangdeyin commented:
@a8m 
> Amazing @huangdeyin! Thanks for the contribution.
> 
> I'll let @zeevmoney also give it a look, but please fix the minor style comments and I think we can merge this.

Thanks for checking, I've fixed these minor issues

a8m commented:
Great @huangdeyin, lint job is failing in the CI. You need to run goimports.

huangdeyin commented:
@a8m 
> Great @huangdeyin, lint job is failing in the CI. You need to run goimports.

I fixed
